### ISSUE TITLE
100-50 equals negative 50

### ISSUE THREAD

**Issue by peter-lernyx** [bdecentgmbh/moodle-enrol_credit#5]
Stefan, potential bug. Moodle 3.8.6, latest plugin version. Phenomenon: Student has 100 credits, course price is 50 credits. Student purchases course, then balance is -50 credits (negative 50) and indeed he is not able to purchase any other courses, getting the message that credit balance is insufficient, so in fact -50 is recorded in the data base. Any ideas?

**Comment by peter-lernyx** [comment]
_can you reproduce this behaviour? Has this (or something similar) happened previously? Feel free to answer either here or on GitHub, whatever you prefer._ 

Hi Stefan, yes this has happened more than once. But it also has shown correct behavior on occasions. I am getting a feeling it has to do with the starting credit and the amount you deduct, to produce an arithmetic error, i.e. 100-50, the result being the same amount that is being deducted, but I will do more tests tomorrow and let you know if you want.

![grafik](https://user-images.githubusercontent.com/75390480/100941974-cf573780-34fa-11eb-90e9-6c0b2f462a9e.png)

**Comment by stefanscholz** [comment]
Ok, please keep me updated. You're the first person to ever report such a behaviour, so let's look into this and figure out why it happens. I'll need details on the exact setup and I'll recreate the environment on my side and try to reproduce it.
Let me know how your tests go and then we can decide next steps.

**Comment by peter-lernyx** [comment]
Sure, can you tell me which table the credits are stored in? I couldn't find it in the Maria database?

**Comment by stefanscholz** [comment]
They are stored in the custom profile field you've created and then configured.  
We do not store it separately.

**Comment by peter-lernyx** [comment]
Yes sure but I can only find the definition of that field in table xxx_user_info_field, I am looking for where the actual value is stored.

**Comment by stefanscholz** [comment]
Should be xxx_user_info_data

**Comment by peter-lernyx** [comment]
got it user_info_data! Thanks, speak soon.
Oh, you were quicker...;-)

**Comment by peter-lernyx** [comment]
Hi Stefan, I think I know what's going on. The module writes an ASCII/Markup string to the data field in _user_info_data in the form of 

![grafik](https://user-images.githubusercontent.com/75390480/101013908-4e815580-3565-11eb-8593-d92ef6145b82.png)

instead of the numerical value "100". That happens when I manipulate the 'credits' user field via the user profile in moodle. I can even purchase something costing 1.000 credits, even when I only have 64 credits on my balance. It doesn't recognize the fact that the value in the data field is in fact invalid.

Whenever I change the credit directly in the database (via phpmyadmin). the modul works just fine.  I am attaching a protocol of my tests. Apologies when it's not pretty.

I will go and see whether in defining the custom field I told it to be a text string, I don't know from memory whether you can actually force this as an arithmetic value in moodle.
[Error Credit Enrolment.pdf](https://github.com/stefanscholz/moodle-enrol_credit/files/5635791/Error.Credit.Enrolment.pdf)

**Comment by peter-lernyx** [comment]
Update: You can force a user field to be of a text but not arithmetic.

![grafik](https://user-images.githubusercontent.com/75390480/101014253-e0895e00-3565-11eb-8cc4-95d16970413e.png)

**Comment by peter-lernyx** [comment]
Yes, that was it. I changed the field from longtext() to MEDIUMINT(9) in the MariaDB and now it works fine. I am curious as to why you did never get this before...

![grafik](https://user-images.githubusercontent.com/75390480/101018385-baff5300-356b-11eb-8155-3040c2c03ccc.png)
